Move voip previews to bottom right corner

JorikSchellekens opened the pull request:
fixes: https://github.com/vector-im/riot-web/issues/14177
relates to: https://github.com/vector-im/riot-web/issues/13635 in PRs for overarching tracking


![audio](https://user-images.githubusercontent.com/23084468/86652909-b0508a00-bfdc-11ea-8149-751f0bac4530.gif)

There is also a video view:

![image](https://user-images.githubusercontent.com/23084468/86653756-569c8f80-bfdd-11ea-954e-7af45f4381c9.png)

JorikSchellekens committed:
Implement incoming call box

turt2live commented on res/css/_components.scss:
```diff
@@ -51,6 +51,7 @@
 @import "./views/avatars/_BaseAvatar.scss";
 @import "./views/avatars/_DecoratedRoomAvatar.scss";
 @import "./views/avatars/_MemberStatusMessageAvatar.scss";
+@import "./views/avatars/_PulsedAatar.scss";
```
spelling

turt2live commented on res/css/views/voip/_CallView2.scss:
```diff
@@ -0,0 +1,94 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
```
```suggestion

// TODO: Rename on launch: https://github.com/vector-im/riot-web/issues/14231

```

turt2live commented on src/components/views/avatars/PulsedAvatar.tsx:
```diff
@@ -0,0 +1,28 @@
+
+const PulsedAvatar: React.FC<IProps> = (props) => {
+    return <div className="mx_PulsedAvatar">
+            {props.children}
```
this is indented too far

turt2live commented on src/components/views/voip/CallContainer.tsx:
```diff
@@ -0,0 +1,37 @@
+
+}
+
+export default class CallContainer extends React.Component<IProps, IState> {
```
```suggestion
export default class CallContainer extends React.PureComponent<IProps, IState> {
```

turt2live commented on src/components/views/voip/CallPreview2.tsx:
```diff
@@ -0,0 +1,126 @@
+import SettingsStore from "../../../settings/SettingsStore";
+
+interface IProps {
+        // A Conference Handler implementation
+        // Must have a function signature:
+        //  getConferenceCallForRoom(roomId: string): MatrixCall
+        ConferenceHandler: any;
```
indentation is off

turt2live commented on src/components/views/voip/CallPreview2.tsx:
```diff
@@ -0,0 +1,126 @@
+            newRoomListActive: SettingsStore.getValue("feature_new_room_list"),
+        };
+
+        SettingsStore.watchSetting("feature_new_room_list", null, (name, roomId, level, valAtLevel, newVal) => this.setState({
+            newRoomListActive: newVal,
+        }));
```
this needs an unwatch too

turt2live commented on src/settings/Settings.js:
```diff
@@ -305,7 +305,7 @@ export const SETTINGS = {
         supportedLevels: LEVELS_ACCOUNT_SETTINGS,
         displayName: _td('Mirror local video feed'),
-        default: false,
+        default: true,
```
This requires a product decision, I think.

JorikSchellekens replied:
How do I go about getting one?

turt2live replied:
making noise in the direction of Nad

turt2live replied:
ftr: https://github.com/vector-im/riot-web/issues/10651 and https://github.com/vector-im/riot-web/issues/5633 are relevant issues.

JorikSchellekens replied:
@nadonomy for context the current 1:1 video calls have a small local video feed. This feed is not mirrored by default although I think it aught to be. (See the issues linked by turt2live)
Currently it looks like this:
![image](https://user-images.githubusercontent.com/23084468/86810851-3cae8b80-c075-11ea-835c-f05e11d27484.png)

But it's probably a lot more intuitive like this:
![image](https://user-images.githubusercontent.com/23084468/86810907-4801b700-c075-11ea-8900-e87d6c9a4fcf.png)

Should we make the default mirrored?

turt2live replied:
Per the call: we'll keep this off by default for now as the screen sharing issue needs work.

JorikSchellekens replied:
Because of https://github.com/vector-im/riot-web/issues/5633 we're not doing this but I'm making it my personal mission to fix that one soon.

turt2live reviewed (changes requested):
Please in the future do not mix TypeScript conversions with functional changes, as it's hard to review. It's made even more difficult to review when all of them are in a single commit.

I've done an initial pass here and it overall seems reasonable, though the tests are super angry.

JorikSchellekens committed:
Merge branch 'develop' of github.com:matrix-org/matrix-react-sdk into joriks/room-list-voip

JorikSchellekens commented:
I thought it would be a small conversion but it got out of hand. I'll be more segmented about it in future.

JorikSchellekens committed:
Fix review details

- spelling mistake
- unwatch watched setting
- lint (indentation)
- use more performant component

JorikSchellekens committed:
lint

JorikSchellekens committed:
Lint whitespaces and semis

JorikSchellekens committed:
Merge branch 'develop' into joriks/room-list-voip

JorikSchellekens committed:
Fix tests

JorikSchellekens committed:
Please fix test

JorikSchellekens committed:
please work

turt2live reviewed (approved):
lgtm assuming product is okay with the change

nadonomy commented:
Just to summarise some internal chat:

- Overall, looks great!
- Mirroring makes sense, both from a UX perspective and from benchmarking other apps (as presumably they've been through the pain on this, and also users would expect our implementation to work in an expected way)
- We need to R&D if we can mirror video feeds without mirroring screen shares

turt2live reviewed (approved):
good to merge once the mirror flip is flipped

JorikSchellekens committed:
Back to defaul orientation
